src/users/email_verifications/mod.rs

```diff
@@ -0,0 +1,95 @@
+use axum::{http::StatusCode, response::IntoResponse};
+use lettre::{
+    message::header::ContentType, transport::smtp::authentication::Credentials, AsyncSmtpTransport,
+    AsyncTransport, Message, Tokio1Executor,
+};
+
+use crate::{ErrorResponse, EMAIL_PASSWORD, EMAIL_SMTP_SERVER, EMAIL_USERNAME};
+
+use self::models::EmailVerification;
+
+pub mod models;
+pub mod routes;
+
+#[derive(thiserror::Error, Debug)]
+pub enum EmailVerificationError {
+    #[error(transparent)]
+    PoolError(#[from] diesel_async::pooled_connection::deadpool::PoolError),
+
+    #[error(transparent)]
+    Diesel(#[from] diesel::result::Error),
+
+    #[error("Email has expired")]
+    ExpiredEmail,
+
+    #[error(transparent)]
+    EmailSendError(#[from] lettre::transport::smtp::Error),
+
+    #[error(transparent)]
+    BodyCreationError(#[from] lettre::error::Error),
+}
+
+impl EmailVerification {
+    async fn send_email(&self, username: String) -> Result<(), EmailVerificationError> {
```

Comment:
should add parameters for the email_username, email_password, and email_smtp_server when moving them to `AppState`